api/repo/update.go

```diff
@@ -170,6 +173,14 @@ func UpdateRepo(c *gin.Context) {
 		r.SetActive(input.GetActive())
 	}
 
+	// set allow events based on input if given
+	if !reflect.DeepEqual(input.GetAllowEvents(), new(library.Events)) {
+		r.SetAllowEvents(input.GetAllowEvents())
```

Comment:
wouldn't we want to compare against r.GetAllowEvents?

Reply:
Hm, I think honestly we just want to compare

```go
if input.AllowEvents != nil
```

Because this line is really just seeing if they supplied something. I don't like using reflect either for something like that

Reply:
sure, that works. i also cringed a little at seeing reflect here :D